src/Zap.sol

```diff
@@ -453,61 +455,80 @@ contract Zap is Reentrancy, Errors {
         requireStage(Stage.LEVEL2)
         returns (uint256 unwound)
     {
-        (
-            uint128 _accountId,
-            uint128 _collateralId,
-            uint256 _collateralAmount,
-            address _collateral,
-            uint256 _zapTolerance,
-            uint256 _unwrapTolerance,
-            uint256 _swapTolerance,
-            /* address _receiver */
-        ) = abi.decode(
+        {
+            (
+                uint128 _accountId,
+                uint128 _collateralId,
+                uint256 _collateralAmount,
+                ,
+                ,
+                uint256 _zapTolerance,
+                uint256 _unwrapTolerance,
+                ,
+            ) = abi.decode(
+                _params,
+                (
+                    uint128,
+                    uint128,
+                    uint256,
+                    address,
+                    bytes,
+                    uint256,
+                    uint256,
+                    uint256,
+                    address
+                )
+            );
+
+            // zap USDC from flashloan into USDx;
+            // ALL USDC flashloaned from Aave is zapped into USDx
+            uint256 usdxAmount = _zapIn(_flashloan, _zapTolerance);
+            // burn USDx to pay off synthetix perp position debt;
+            // debt is denominated in USD and thus repaid with USDx
+            _burn(usdxAmount, _accountId);
+
+            /// @dev given the USDC buffer, an amount of USDx
+            /// necessarily less than the buffer will remain (<$1);
+            /// this amount is captured by the protocol
+
+            // withdraw synthetix perp position collateral to this contract;
+            // i.e., # of sETH, # of sUSDe, # of sUSDC (...)
+            _withdraw(_collateralId, _collateralAmount, _accountId);
+
+            // unwrap withdrawn synthetix perp position collateral;
+            // i.e., sETH -> WETH, sUSDe -> USDe, sUSDC -> USDC (...)
+            unwound =
+                _unwrap(_collateralId, _collateralAmount, _unwrapTolerance);
+
+            // establish total debt now owed to Aave;
+            // i.e., # of USDC
+            _flashloan += _premium;
+        }
+
+        (,,, address _collateral, bytes memory _path,,, uint256 _swapTolerance,)
+        = abi.decode(
```

Comment:
why are we decoding this again? was there a stack too deep issue?

Reply:
yea exactly